### PR TITLE
[feature-wip](multi-catalog) support partition&missing columns in parquet lazy read

### PR DESCRIPTION
**Pull request by AshinGau**
# Proposed changes
Support partition&missing columns in parquet lazy read.

## Problem summary
PR https://github.com/apache/doris/pull/13917 has supported lazy read for non-predicate columns in ParquetReader, but can't trigger lazy read when predicate columns are partition or missing columns. This PR support such case, and fill partition and missing columns in `FileReader`.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/text_converter.h [hunk 1]
```diff
@@ -54,7 +54,8 @@ class TextConverter {
                       const char* data, size_t len, bool copy_string, bool need_escape);
 
     bool write_vec_column(const SlotDescriptor* slot_desc, vectorized::IColumn* nullable_col_ptr,
-                          const char* data, size_t len, bool copy_string, bool need_escape);
+                          const char* data, size_t len, bool copy_string, bool need_escape,
+                          size_t rows = 1);
```
Comment (morningman) [inline be/src/exec/text_converter.h:58]: add comment in code to explain this magic num

File: be/src/exec/text_converter.hpp [hunk 2]
```diff
@@ -191,16 +191,18 @@ inline bool TextConverter::write_column(const SlotDescriptor* slot_desc,
...
     if (slot_desc->is_nullable()) {
         auto* nullable_column = reinterpret_cast<vectorized::ColumnNullable*>(nullable_col_ptr);
         if ((len == 2 && data[0] == '\\' && data[1] == 'N') || len == SQL_NULL_DATA) {
-            nullable_column->insert_data(nullptr, 0);
+            nullable_column->insert_many_defaults(rows);
             return true;
         } else {
-            nullable_column->get_null_map_data().push_back(0);
+            auto& null_map = nullable_column->get_null_map_data();
+            null_map.resize_fill(null_map.size() + rows, 0);
```
Comment (morningman) [inline be/src/exec/text_converter.hpp:205]: You need to modify the `if` code block in L333

File: be/src/vec/exec/format/generic_reader.h [hunk 3]
```diff
@@ -43,6 +44,18 @@ class GenericReader {
         return Status::NotSupported("get_parser_schema is not implemented for this reader.");
     }
     virtual ~GenericReader() = default;
+
+    bool fill_all_columns() const { return _fill_all_columns; }
```
Comment (morningman) [inline be/src/vec/exec/format/generic_reader.h:48]: Add comment for interface.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.25 seconds
 load time: 461 seconds
 storage size: 17180196107 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221114171928_clickbench_pr_45924.html

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**